canary/src/main/java/io/littlehorse/canary/aggregator/topology/TaskRunLatencyTopology.java

```diff
@@ -0,0 +1,44 @@
+package io.littlehorse.canary.aggregator.topology;
+
+import io.littlehorse.canary.aggregator.serdes.ProtobufSerdes;
+import io.littlehorse.canary.proto.Metric;
+import io.littlehorse.canary.proto.MetricAverage;
+import java.time.Duration;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.kstream.Suppressed;
+import org.apache.kafka.streams.kstream.TimeWindows;
+
+@Slf4j
+public class TaskRunLatencyTopology {
+
+    public TaskRunLatencyTopology(final KStream<String, Metric> metricStream) {
+        metricStream
+                .filter((key, value) -> value.hasTaskRunLatency())
+                .groupByKey()
+                .windowedBy(TimeWindows.ofSizeAndGrace(Duration.ofMinutes(1), Duration.ofSeconds(30)))
```

Comment:
Question here—I haven't done a full deep dive into the code. How do we want to handle grace periods? I am wondering about the specific case when:
* A `TaskRun` is scheduled.
* Before the Task Worker starts the task, the server crashes and goes through a super-long state restoration
* One hour later, the `TaskRun` is executed.

Can we support this case?